src/search.ts

```diff
@@ -354,7 +362,8 @@ function createSearchPanel(view: EditorView) {
 function defaultQuery(state: EditorState, fallback?: Query) {
   let sel = state.selection.main
   let selText = sel.empty || sel.to > sel.from + 100 ? "" : state.sliceDoc(sel.from, sel.to)
-  return fallback && !selText ? fallback : new StringQuery(selText.replace(/\n/g, "\\n"), "", fallback?.caseInsensitive || false)
```

Comment:
I think you want `??` instead of `||` here, so that if `fallback` is given its setting is preserved.

Reply:
Yes! ?? is right.

It's such a nice operator :)